common4j/src/main/com/microsoft/identity/common/java/cache/HttpCache.java

```diff
@@ -0,0 +1,59 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.java.cache;
+
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+public class HttpCache {
```

Comment:
javadoc

Reply:
Really thinking we need to hide this in the client, and it needs to be non-static.  Since we're using factories to access things anyway, should we have a static platform factory class as well as the *Components classes?

Reply:
the way this is being used is similar to Logger, Telemetry, Device class, as in, they are being called out of nowhere.

Having to pass a factory class everywhere would make them harder to use.

We could have a static platform factory class, but I prefer having only 2 patterns instead of 3. If we are going that route, then we should move Logger, Telemetry, Device to that one as well.... probably something to think about post P0.

Reply:
https://identitydivision.visualstudio.com/Engineering/_queries/edit/1446143/?triage=true&workitem=1485244